### PR TITLE
Add .editorconfig

### LINKED ISSUE #7: add .editorconfig
Store a canonical `.editorconfig` in this repo so that everyone who commits code follows the same style conventions.
1. copy https://gist.github.com/rdmurphy/72a3d1e2f89a0e909c10
2. add a comment at the top, something like:
   
   ```
   # add this to your project with:
   #     wget raw.github/path/.editorconfig
   # any changes to this file should be merged back into the master styleguides repo
   ```

### PR DESCRIPTION
**Pull request by crccheck**
Intended to be shared between everyone.

closes #7

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+[*]
+indent_style = space
+indent_size = 2
+end_of_line = lf
+charset = utf-8
+trim_trailing_whitespace = true
+insert_final_newline = true
+
+[*.md]
+trim_trailing_whitespace = false
```
Comment (crccheck) [inline .editorconfig:26]: why don't you trim trailing whitespace in markdown files, @texastribune/news-apps ?

Comment (rdmurphy) [inline .editorconfig:26]: Because of how Markdown [uses white-space](https://twitter.com/beyondsanity/status/397728531179786240) to insert `<br>`'s.

**Comment by rdmurphy** [comment]
Just added a line for `Makefile`'s. Was messing with the one on salaries and had Editorconfig converting tabs to spaces bite me.

https://github.com/texastribune/styleguides/commit/ae9a389b8ddc7a1d665a9eaab0693d9d94d4349c

**Comment by crccheck** [comment]
ok haven't heard anything in a while. merging.

**Pull request merged**